src/Database/Schema/Migrations/Filesystem.hs

```diff
@@ -104,8 +106,8 @@ migrationFromPath path = do
     readMigrationFile = do
       ymlExists <- doesFileExist (addNewMigrationExtension path)
       if ymlExists
-        then parseYamlFile (addNewMigrationExtension path) `catch` (\(e::IOException) -> throwFS $ show e)
-        else parseYamlFile (addMigrationExtension path filenameExtensionTxt) `catch` (\(e::IOException) -> throwFS $ show e)
+        then decodeFileThrow (addNewMigrationExtension path) `catch` (\(e::IOException) -> throwFS $ show e)
```

Comment:
The old code caught `IOException`. Is that how the new Yaml parser behaves as well?

Reply:
On invalid YAML it throws `ParseException`. I don't know what the old parser did on invalid YAML as I never got it to work. Assuming it threw `IOException` on invalid YAML and this needs to be caught and rethrown using `throwFS` I'm adding a catch for `ParseException`. As for actual IO errors, e.g. file permissions problems, the new parser seems to throw `ParseException` too so catching `IOException` is probably not needed at all.